File: hw/Vaca/u3/HW30ObserverAndStrategy/HW30StrategyJavaScript/controller/SortingContext.js
export class SortingContext {
    setSortStrategy(strategy) {
        this.strategy = strategy;
    }

    sort(numbers) {
        if (this.strategy) {
            this.strategy.sort(numbers);
        } else {
            throw new Error("Sorting strategy not set");
        }
    }

    getSortStrategyName() {
        if (this.strategy) {
            return this.strategy.getStrategyName();
        } else {
            return "No strategy set";
        }
    }
}

File: hw/Vaca/u3/HW30ObserverAndStrategy/HW30StrategyJava/src/main/java/ec/edu/espe/controller/SortingContext.java
package ec.edu.espe.controller;

/**
 *
 * @author MSI-PULSE
 */
public class SortingContext {
    private SortingStrategy strategy;

    public void setSortStrategy(SortingStrategy strategy) {
        this.strategy = strategy;
    }

    public void sort(int[] numbers) {
        if (strategy != null) {
            strategy.sort(numbers);
        } else {
            throw new IllegalStateException("Sorting strategy not set");
        }
    }

    public String getSortStrategyName() {
        if (strategy != null) {
            return strategy.getStrategyName();
        } else {
            return "No strategy set";
        }
    }
}
